### PR TITLE
chore: update prettier to `3.0`

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

It updates prettier to `3.0`

## Testing

I run locally the script that formats the code, and it works as expected.

## Docs

N/A

### CODE REVIEW

File: .prettierrc.cjs [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
   tabWidth: 2,
   trailingComma: 'es5',
   useTabs: true,
-  plugins: ['./node_modules/prettier-plugin-astro'],
+  plugins: ['./node_modules/prettier-plugin-astro/dist/index.js'],
```
Comment (bluwy) [inline .prettierrc.cjs:9]: I think we can also simply specify `plugins: ['prettier-plugin-astro']` in Prettier 3

File: packages/astro/astro-jsx.d.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@
  * Adapted from React’s TypeScript definition from DefinitelyTyped.
  * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts
  */
-declare namespace astroHTML.JSX {
+namespace astroHTML.JSX {
```
Comment (bluwy) [inline packages/astro/astro-jsx.d.ts:12]: Is the `declare` removal caused by prettier?

File: .prettierrc.cjs (hunk 1 above)
Comment (Princesseuh) [reply .prettierrc.cjs:9]: Yes, I would recommend to do this actually, it's better

File: packages/astro/astro-jsx.d.ts (hunk 2 above)
Comment (ematipico) [reply packages/astro/astro-jsx.d.ts:12]: Yes. The files changed are caused by running prettier locally (which is good, it seems we caught a bug?)

**Review by bluwy — APPROVED** [review]

**Pull request merged**